src/App.tsx

```diff
@@ -16,77 +25,128 @@ const stationMap = new Map<string, string>([
 
 const stationNames = Array.from(stationMap.keys());
 
-const urlMaker = (stationOne: string, stationTwo: string) => {
-    return `https://www.lner.co.uk/travel-information/travelling-now/live-train-times/depart/${
-        stationMap.get(stationOne)}/${stationMap.get(stationTwo)}/`;
+const apiUrlMaker = (parameters: IApiUrlMaker) => {
+    const extractedParameters = Array.from(Object.entries(parameters));
+    return 'https://mobile-api-softwire2.lner.co.uk/v1/fares?'
+        + extractedParameters.map(([k, v]) => `${k}=${v}`).join('&');
 };
 
 const App = () => {
 
-    const [departureStation, setDepartureStation] = React.useState('');
-    const [arrivalStation, setArrivalStation] = React.useState('');
+    const [originStationName, setOriginStationName] = React.useState('');
+    const [destinationStationName, setDestinationStationName] = React.useState('');
     const [disableSubmit, setDisableSubmit] = React.useState(true);
+    const [trainData, setTrainData] = React.useState(
+        null as null | {'outboundJourneys': JourneyEntry[]},
+    );
```

Comment:
Have you thought about using this kind of syntax?
const [destinationStation, setDestinationStation] = useState<string | null>(null);